lib/ecto/embedded.ex

```diff
@@ -121,6 +121,51 @@ defmodule Ecto.Embedded do
     :error
   end
 
+  @impl Ecto.ParameterizedType
+  def cast_with_current(%{action: :ignore}, _, %{cardinality: :one}), do: :ignore
+  def cast_with_current(value, current, %{on_cast: on_cast, related: related, cardinality: cardinality} = embedded) do
+    value = if cardinality == :many and is_list(value) do
+      Enum.reject(value, &match?(%{action: :ignore}, &1))
+    else
+      value
+    end
+    on_cast = case on_cast do
+      nil -> on_cast_default(related)
+      _ -> on_cast
+    end
+    case Relation.cast(embedded, nil, value, current, on_cast) do
+      {:ok, change, valid?} -> {:ok, change, valid?}
+      {:error, {message, _meta}} -> {:error, message: message}
```

Comment:
```suggestion
      {:error, {message, _meta}} -> {:error, [message: message] ++ meta}
```

In fact, I would even change `Relation.cast` to return `{:error, [message: message] ++ meta}` directly and then we do `Keyword.pop` in `cast_relation` to get the message.